src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -239,7 +238,13 @@ void LambdaFormInvokers::dump_static_archive_invokers() {
 void LambdaFormInvokers::read_static_archive_invokers() {
   if (_static_archive_invokers != nullptr) {
     for (int i = 0; i < _static_archive_invokers->length(); i++) {
-      Array<char>* line = _static_archive_invokers->at(i);
+      u4 offset = _static_archive_invokers->at(i);
+      Array<char>* line;
+      if (ArchiveBuilder::is_active()) {
+        line = ArchiveBuilder::current()->offset_to_buffered<Array<char>*>(offset);
+      } else {
```

Comment:
The `if (ArchiveBuilder::is_active()` branch doesn't seem necessary, as this function is called in only a single place where this condition is false.